### PR TITLE
Fix compilation with ifort

### PR DESCRIPTION
**Pull request by danielhollas**
These are all ugly hacks for now, but I don't have time to fix these things properly for now.
Long term we need to get rid of all GNU/GCC-specific stuff altogether.

### CODE REVIEW

File: dev_scripts/setup_dev_environment.sh [hunk 1]
```diff
@@ -11,14 +11,16 @@ set -euxo pipefail
 CWD=`dirname $0`
 if [[ ! -f $CWD/../.git/hooks/pre-push ]];then
     cp $CWD/pre-push.hook $CWD/../.git/hooks/pre-push
-    chmod +x ../.git/hooks/pre-push
+    chmod +x $CWD/../.git/hooks/pre-push
 fi
 
 # Heuristic: if .vimrc exists, we presume we're dealing with a VIM user :-)
-if [[ -f /home/$USER/.vimrc ]];then
+if [[ -f ~/.vimrc ]];then
```
Comment (danielhollas) [inline dev_scripts/setup_dev_environment.sh:18]: Durham clusters don't have `/home/$USER/` directory. :-)

File: src/init.F90 [hunk 2]
```diff
@@ -1107,6 +1123,8 @@ subroutine print_logo()
       print '(a)', ' '
    end subroutine print_logo
 
+end subroutine init
```
Comment (danielhollas) [inline src/init.F90:1126]: `print_runtime_info` has a system call to `ldd`. The fortran compiler is not happy with the fact that we also have a namelist called `system`. Putting `print_runtime_info()` outside of `init()` fixes that.

File: water_potentials/Makefile [hunk 3]
```diff
@@ -1,4 +1,3 @@
-CXX = g++
```
Comment (danielhollas) [inline water_potentials/Makefile:1]: This should not be hardcoded, but defined in `make.vars`

**Comment by danielhollas** [comment]
@suchanj can you take a look? btw: Big congrats on the poster prize! :tada: 

I think I'll be submitting more PRs in the next few weeks. LMK if you're down for reviewing them, or whether I should merge them without review. Thanks!

**Comment by suchanj** [comment]
@danielhollas Thanks! Regarding the new PRs - I'd like to stay in the loop but don't want to hinder the merging process by taking time. So I would propose that you merge whatever you need, but do not immediately delete the branches from repository so they would still be visible :+1:

**Review by suchanj — APPROVED** [review]

**Comment by danielhollas** [comment]
@suchanj awesome, thanks! Sound good. I think some of them will be just some minor refactoring and cleanups. For anything bigger I'll always wait for you to take a look.

**Pull request merged**